File: plugins/flutter_assurance/android/src/main/java/com/adobe/marketing/mobile/flutter/FlutterAssurancePlugin.java
/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

package com.adobe.marketing.mobile.flutter;

import android.util.Log;

import com.adobe.marketing.mobile.Assurance;

import androidx.annotation.NonNull;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;

/** FlutterAssurancePlugin */
public class FlutterAssurancePlugin implements FlutterPlugin, MethodCallHandler {

  private static final String TAG = "FlutterAssurancePlugin";

  private MethodChannel channel;

  @Override
  public void onAttachedToEngine(@NonNull final FlutterPluginBinding binding) {
    channel = new MethodChannel(binding.getBinaryMessenger(), "flutter_assurance");
    channel.setMethodCallHandler(new FlutterAssurancePlugin());
  }

  @Override
  public void onDetachedFromEngine(@NonNull final FlutterPluginBinding binding) {
    if (channel != null) {
      channel.setMethodCallHandler(null);
    }
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if ("extensionVersion".equals(call.method)) {
      result.success(Assurance.extensionVersion());
    } else if ("startSession".equals(call.method)) {
      String url = (String) call.arguments;
      if (url.isEmpty()) {
        Log.e(TAG, "Unable to start assurance session, argument parsing failed");
        return;
      }
      Assurance.startSession(url);
      result.success(null);
    } else {
      result.notImplemented();
    }
  }


}

File: plugins/flutter_assurance/ios/Classes/FlutterAssurancePlugin.m
/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

#import "FlutterAssurancePlugin.h"
#import "AEPAssurance.h"
#import "ACPCore.h"

@implementation FlutterAssurancePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_assurance"
            binaryMessenger:[registrar messenger]];
  FlutterAssurancePlugin* instance = [[FlutterAssurancePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"extensionVersion" isEqualToString:call.method]) {
      result([AEPAssurance extensionVersion]);
  } else if ([@"startSession" isEqualToString:call.method]) {
      NSString *url = (NSString *) call.arguments;
      [AEPAssurance startSession:[NSURL URLWithString: url]];
      result(nil);
  } else {
      result(FlutterMethodNotImplemented);
  }
}

@end
